Google.Api.Generator.Rest/Models/DataPropertyModel.cs

```diff
@@ -114,6 +115,63 @@ private IEnumerable<MemberDeclarationSyntax> GenerateDateTimeProperties(SourceFi
                 .WithSetBody(rawProperty.Assign(ctx.Type(typeof(Utilities)).Call(nameof(Utilities.GetStringFromDateTime))(valueParameter)));
         }
 
+        private IEnumerable<MemberDeclarationSyntax> GenerateGoogleDateTimeProperties(SourceFileContext ctx)
+        {
+            if (_schema.Type != "string" || _schema.Required == true || _schema.Properties is object ||
+                _schema.AdditionalProperties is object || _schema.Repeated == true ||
+                _schema.Enum__ is object || _schema.Ref__ is object)
+            {
+                throw new ArgumentException("Unable to handle complex google-datetime properties");
+            }
+
+            // For google-datetime properties, we generate:
+            // - Three properties: Xyz (object), XyzRaw (string), XyzDateTimeOffset (DateTimeOffset?)
+            // - Two fields, to back Xyz and XyzRaw
+            //
+            // Each property setter will set both fields.
+
+            // The type of "value" is irrelevant to our uses of this, so it's okay that in the raw property it should be a string.
+            var valueParameter = Parameter(ctx.Type<object>(), "value");
+            var rawField = Field(Modifier.Private, ctx.Type<string>(), $"_{Name}Raw");
+            var objectField = Field(Modifier.Private, ctx.Type<object>(), $"_{Name}");
+
+            // TODO: Use nameof
```

Comment:
nit: Can't remember what this was waiting on?

Reply:
Oh, done now - thanks!